File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO2/StrategyFull/src/Sportsman.java
public class Sportsman{
    private final String name;
    private final SportsmanStrategyRun strategyRun;
    private final SportsmanStrategySwim strategySwim;
    private final SportsmanStrategyJump strategyJump;
    public Sportsman(String name,SportsmanStrategyRun strategyRun,SportsmanStrategySwim strategySwim,SportsmanStrategyJump strategyJump) {
        this.name = name;
        this.strategyRun = strategyRun;        
        this.strategySwim = strategySwim;
        this.strategyJump = strategyJump;
    }
    public String run() {return "Running";}  
    public String swim() {return "Swimming";}
    public String jump() {return "Jumping";}
    public String execute(SportsmanCommand command){
        return name+": " + command.execute(this);
    }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2018/B/CommandStrategyTest/CommandStrategyTest/Sportsman.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommandStrategyTest
{
    class Sportsman
    {
        public string name { get; set; }
        private StrategyRun run;
        StrategyJump jump;
        StrategySwim swim;
        public Sportsman(string name,StrategyRun run, StrategyJump jump, StrategySwim swim)
        {
            this.run = run;
            this.jump = jump;
            this.swim = swim;
            this.name = name;
        }
        public string Run() { return run.execute(); }
        public string Jump() { return jump.execute(); }
        public string Swim() { return swim.execute(); }
    }
}
